neurodsp/sim/combined.py

```diff
@@ -82,3 +84,89 @@ def sim_combined(n_seconds, fs, components, component_variances=1):
     sig = np.sum(sig_components, axis=0)
 
     return sig
+
+
+@normalize
+def sim_central_freq(n_seconds, fs, central_freq, bw, ht, ap_func='sim_powerlaw',
+                     ap_kwargs={'exponent': -2}):
+    """
+    Returns a time series whose full power spectrum consists of an aperiodic component
+    and a gaussian peak at central_freq with standard deviation bw and relative height ht.
+
+
+    Parameters
+    -----------
+    n_seconds: float
+        Number of seconds elapsed in the time series.
+    fs: float
+        Sampling rate.
+    central_freq: float
+        Central frequency for the gaussian peak in Hz.
+    bw: float
+        Bandwidth, or standard deviation, of gaussian peak in Hz.
+    ht: float
+        Relative height in log_10(Hz) over the aperiodic component of the gaussian peak at
+        central_freq.
+    ap_func : str, optional, default: 'sim_powerlaw'
+        An aperiodic simulation function from ``neurodsp.sim.aperiodic``.
+    ap_kwargs : dict, optional, default: {'exponent': -2}
+        Keyword arguments to use with ``ap_func``.
+
+    Returns
+    -------
+    sig: 1d array
+        Time series with desired power spectrum.
+
+    Examples
+    --------
+    Simulate an aperiodic component with exponent -2 superimposed over
+    an oscillatory component with central frequency 20.
+
+    >>> sig = sim_central_freq(n_seconds=50, fs=500, central_freq=20, bw=5, ht=7,
+    ...                        ap_func='sim_powerlaw', ap_params={'exponent': -2})
+    """
+
+    times = create_times(n_seconds, fs)
+
+    # Construct the aperiodic component and compute its Fourier transform. Only use the
+    # first half of the frequencies from the Fourier transform since the signal is real.
+    sig_components = {ap_func: ap_kwargs}
+    sig_ap = sim_combined(n_seconds, fs, sig_components)
```

Comment:
This doesn't _really_ matter, but I don't think we need to call `sim_combined` here, since we are not actually combining signals. `ap_func` should be a callable, so this should work as:
`sig_ap = ap_func(n_seconds, fs, **ap_kwargs)`

Reply:
Ahh, okay, so I just noticed that the function is being passed in as a string label, which totally makes sense, but we  can still skip the `sim_combined` step, with this (basically, doing what `sim_combined` does to go from label -> callable:
```
from neurodsp.sim.info import get_sim_func
ap_func = get_sim_func(ap_func) if isinstance(ap_func, str) else ap_func) 
sig_ap = ap_func(n_seconds, fs, **ap_kwargs)
```

Side benefit: this actually generalizes so that `ap_func` can be a string label for a function within the module, or be _any callable_ (so one could pass in custom functions, as long as they return a signal, as expected).

Reply:
Quick comment: add support for array input instead.

Reply:
Just making sure, but replace `ap_func` and `ap_kwargs`, with something like `ap_sig` instead?

Reply:
I think that makes sense. We would need to do additional checks to ensure the `n_seconds` and `fs` are consistent with `ap_sig` to make sure it can be cleanly add to the periodic signal.

Reply:
Array support added instead.